internal/controllers/machinedeployment/machinedeployment_sync.go

```diff
@@ -257,15 +257,8 @@ func (r *Reconciler) computeDesiredMachineSet(ctx context.Context, deployment *c
 		name = existingMS.Name
```

Comment:
```
// Add foregroundDeletion finalizer to MachineSet if the MachineDeployment has it.
if sets.New[string](deployment.Finalizers...).Has(metav1.FinalizerDeleteDependents) {
    finalizers = []string{metav1.FinalizerDeleteDependents}
}
```
in l.234

I think this code is basically unreachable. Because we never create a MachineSet after the deletionTimestamp is already set. And the finalizer and the deletionTimestamp seem to be set by the kube-apiserver through the delete call at the same time:

> The Kubernetes API server sets the object's metadata.deletionTimestamp field to the time the object was marked for deletion.
> * The Kubernetes API server also sets the metadata.finalizers field to foregroundDeletion.
> * The object remains visible through the Kubernetes API until the deletion process is complete.
> https://kubernetes.io/docs/concepts/architecture/garbage-collection/#foreground-deletion

Reply:
Done

Reply:
wouldn't this bit of code ensure that when you are removing a MD with foreground, then foreground is also honoured between MS deletion and child Machines?

Reply:
I think it doesn't change that behavior. I did the following test with this PR:
* Create Cluster & MD
* Add additional "test" finalizer to MS & Machine
* Delete MD with foreground
* Check that:
  * MD gets the foregroundDeletion finalizer + deletionTimestamp
  * MS gets the foregroundDeletion finalizer + deletionTimestamp
  * Machine gets the deletionTimestamp

Then I:
* Removed test finalizer from Machine => Machine went away
* Removed test finalizer from MachinSet => MachineSet and then MachineDeployment went away

Please note that computeDesiredMachineSet is not executed anymore after the deletionTimestamp has been set on MachineDeployment (because we run reconcileDelete instead)